src/core/net/srp_server.hpp

```diff
@@ -521,28 +522,30 @@ class Server : public InstanceLocator, private NonCopyable
         friend class LinkedListEntry<UpdateMetadata>;
 
     public:
-        static UpdateMetadata *  New(Instance &               aInstance,
-                                     const Dns::UpdateHeader &aHeader,
-                                     Host *                   aHost,
-                                     const Ip6::MessageInfo & aMessageInfo);
-        void                     Free(void);
-        TimeMilli                GetExpireTime(void) const { return mExpireTime; }
-        const Dns::UpdateHeader &GetDnsHeader(void) const { return mDnsHeader; }
-        Host &                   GetHost(void) { return *mHost; }
-        const Ip6::MessageInfo & GetMessageInfo(void) const { return mMessageInfo; }
-        bool                     Matches(const Host *aHost) const { return mHost == aHost; }
+        static UpdateMetadata *    New(Instance &               aInstance,
+                                       const Dns::UpdateHeader &aHeader,
+                                       Host *                   aHost,
+                                       const Ip6::MessageInfo & aMessageInfo);
+        void                       Free(void);
+        TimeMilli                  GetExpireTime(void) const { return mExpireTime; }
+        const Dns::UpdateHeader &  GetDnsHeader(void) const { return mDnsHeader; }
+        otSrpServerServiceUpdateId GetId(void) const { return mId; }
+        Host &                     GetHost(void) { return *mHost; }
+        const Ip6::MessageInfo &   GetMessageInfo(void) const { return mMessageInfo; }
+        bool                       Matches(otSrpServerServiceUpdateId aId) const { return mId == aId; }
 
     private:
         UpdateMetadata(Instance &               aInstance,
                        const Dns::UpdateHeader &aHeader,
                        Host *                   aHost,
                        const Ip6::MessageInfo & aMessageInfo);
 
-        TimeMilli         mExpireTime;
-        Dns::UpdateHeader mDnsHeader;
-        Host *            mHost;        // The host will be updated. The UpdateMetadata has no ownership of this host.
-        Ip6::MessageInfo  mMessageInfo; // The message info of the DNS update request.
-        UpdateMetadata *  mNext;
+        TimeMilli                  mExpireTime;
+        Dns::UpdateHeader          mDnsHeader;
+        otSrpServerServiceUpdateId mId;   // The ID of this service update transaction.
```

Comment:
Suggestion: we can add `typedef otSrpServerServiceUpdateId ServiceUpdateId` at the top of the `Server` class to mirror the type and then use it inside (helps make the type name shorter).